tests/spec/filter/test_take.py

```diff
@@ -54,3 +56,49 @@ def test_take_with_expr(spec_test):
             3: None,
         },
     )
+
+
+@pytest.mark.xfail_in_memory
+def test_chain_multiple_takes(spec_test):
+    table_data = {
+        e: """
+              | i1 | b1
+            --+----+----
+            1 | 1  | T
+            1 | 2  | T
+            1 | 3  | F
+            """,
+    }
+
+    spec_test(
+        table_data,
+        e.take(e.i1 >= 2).take(e.b1).i1.sum_for_patient(),
+        {
+            1: 2,
+        },
+    )
+
+
+# This is here (temporarily?) to capture the constructs which the in-memory engine can
+# currently handle correctly
+def test_multiple_takes_without_chaining(spec_test):
+    table_data = {
+        e: """
+              | i1 | b1
+            --+----+----
+            1 | 1  | T
+            1 | 2  | T
+            1 | 3  | F
+            """,
+    }
+
+    f1 = e.take(e.i1 >= 2)
+    f2 = f1.take(f1.b1)
+
+    spec_test(
+        table_data,
+        f2.i1.sum_for_patient(),
+        {
+            1: 2,
+        },
+    )
```

Comment:
Yeah I agree. But I want to run these against both SQLite and the in-memory engine and all the machinery for doing that is currently in `specs/conftest.py`. Obviously this could be refactored, but there's only so high you can pile the yaks.

Reply:
Yes, understood. As long as you agree it's not idea, that's fine.

Reply:
I'll stick a TODO marker on the comment so I can find these later